### PR TITLE
doc: added documentation for input type=file

### PR DESCRIPTION
**Pull request by vishalvrv9**
Have added some explanation on how to implement the form input type=file in React as suggested in #86. Please review and revert :smile: 

Have used a ref to the file input and used it to access the file(s) in submit handler instead of saving the file in the state. (Was suggested as a better solution in #218)

### CODE REVIEW

**Comment by vishalvrv9** [comment]
@bvaughn 
Was wondering if this PR works?

File: content/docs/forms.md [hunk 1]
```diff
@@ -189,6 +189,46 @@ Overall, this makes it so that `<input type="text">`, `<textarea>`, and `<select
...
+        </label>
+        <br/>
+        <button type='submit'>Submit</button>
+      </form>
+    );
+  }
+}
+```
+
+[Try it on CodePen.](https://codepen.io/fetard/pen/bYvaJY?editors=0010)
```
Comment (bvaughn) [inline content/docs/forms.md]: Add this code to the [`examples` directory](https://github.com/reactjs/reactjs.org/tree/master/examples) and point to either Codepen or CodeSandbox using the [`gatsby-remark-code-repls` plugin](https://www.gatsbyjs.org/packages/gatsby-remark-code-repls/). 😄  This way it's easier going forward for others to help maintain the demo.

Comment (vishalvrv9) [reply content/docs/forms.md]: Have been travelling. Will do that within 48 hours. :smile:

**Comment by vishalvrv9** [comment]
Have committed the changes in this branch as suggested in the review @bvaughn

File: examples/components-and-props/input-type-file.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+class FileInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit (event) {
```
Comment (bvaughn) [inline examples/components-and-props/input-type-file.js]: Assuming you are interested in going the `embed` route I mentioned above, you can add line-highlighting into the file by adding a comment above `handleSubmit` like so:
>  // highlight-range{1-4}

File: examples/components-and-props/input-type-file.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    event.preventDefault();
+    alert(`Selected file - ${this.fileInput.files[0].name}`);
+  }
+
+  render() {
+    return (
+      <form onSubmit={this.handleSubmit}>
+        <label>
+          Upload file:
+          <input type='file' ref={input => {this.fileInput = input}} />
```
Comment (bvaughn) [inline examples/components-and-props/input-type-file.js]: And another above `<input... />` like so:

> {/* highlight-next-line */}

**Comment by vishalvrv9** [comment]
@bvaughn Can't exactly figure out what went wrong. Was wondering if you could help resolve? :disappointed_relieved:

**Comment by bvaughn** [comment]
Yup! Looks like CI is failing on the `yarn prettier` command. Run it locally and you'll see that the example gets slightly reformatted.

Since this reformats the example slightly, you'll want to verify the line number highlights. To make sure Gatsby regenerates the markdown for this example, the safest thing to do is clear cache before spinning up the devserver: `rm -rf .cache && yarn dev`. (This is a downside of using this plug-in and maybe something I can address with a future update...)

Anyway, the end result after doing both of the following should be something like this:
```diff
diff --git a/examples/forms/input-type-file.js b/examples/forms/input-type-file.js
index 5cc084d9f..5eff7c5b8 100644
--- a/examples/forms/input-type-file.js
+++ b/examples/forms/input-type-file.js
@@ -1,27 +1,45 @@
 class FileInput extends React.Component {
   constructor(props) {
     super(props);
-    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(
+      this
+    );
   }
-// highlight-range{1-4}
-  handleSubmit (event) {
+
+  // highlight-range{5}
+  handleSubmit(event) {
     event.preventDefault();
-    alert(`Selected file - ${this.fileInput.files[0].name}`);
+    alert(
+      `Selected file - ${
+        this.fileInput.files[0].name
+      }`
+    );
   }
 
   render() {
     return (
-      <form onSubmit={this.handleSubmit}>
+      <form
+        onSubmit={this.handleSubmit}>
         <label>
           Upload file:
-          {/* highlight-next-line */}
-          <input type='file' ref={input => {this.fileInput = input}} />
+          {/* highlight-range{1-6} */}
+          <input
+            type="file"
+            ref={input => {
+              this.fileInput = input;
+            }}
+          />
         </label>
-        <br/>
-        <button type='submit'>Submit</button>
+        <br />
+        <button type="submit">
+          Submit
+        </button>
       </form>
     );
   }
 }
 
-ReactDOM.render(<FileInput />, document.getElementById('root'));
+ReactDOM.render(
+  <FileInput />,
+  document.getElementById('root')
+);
```

**Comment by vishalvrv9** [comment]
@bvaughn That fixed it. Thanks. :smile: (Apologies for not being able to figure out the CI problem myself)

**Comment by vishalvrv9** [comment]
Immensely thankful @bvaughn for your mentorship towards my first contribution. ✌️

**Pull request merged**